bot/management/commands/bot.py

```diff
@@ -26,46 +26,67 @@ async def repost(update, context):
             inline_keyboard = update.channel_post.reply_markup
         else:
             work_content = None
+            inline_keyboard = update.channel_post.reply_markup
 
         output_channel = await OutputChannel.objects.aget(pk=channel_binding.output_channel_id)
         if is_text_only:
             content = await sync_to_async(update_content)(channel_binding, work_content)
+            markup = await sync_to_async(update_markup)(channel_binding, inline_keyboard)
             await context.bot.send_message(
                 chat_id=output_channel.telegram_id,
                 text=content,
                 parse_mode="HTML",
-                reply_markup=inline_keyboard
+                reply_markup=markup
             )
 
         elif is_text_with_image:
             content = await sync_to_async(update_content)(channel_binding, work_content)
+            markup = await sync_to_async(update_markup)(channel_binding, inline_keyboard)
             await context.bot.copy_message(
                 chat_id=output_channel.telegram_id,
                 message_id=update.effective_message.id,
                 from_chat_id=update.effective_chat.id,
                 caption=content,
-                parse_mode="HTML"
+                parse_mode="HTML",
+                reply_markup=markup
             )
 
         else:
             await context.bot.copy_message(
                 chat_id=output_channel.telegram_id,
                 message_id=update.effective_message.id,
-                from_chat_id=update.effective_chat.id
+                from_chat_id=update.effective_chat.id,
+                reply_markup=markup
             )
 
 
+def update_markup(channel, inline_keyboard):
+    markup = inline_keyboard
+
+    if markup is not None:
+        for items in markup['inline_keyboard']:
+            for item in items:
+                for username_replacement in channel.username_replacements.all():
+                    if 'Vijaysignal' in item['url']:
```

Comment:
Looks like a hardcoded value. Such errors should never reach code review. You should review a PR request every time you make it.